### PR TITLE
feat: Add pagination list_network_requests

### PR DESCRIPTION
**Pull request by OzTamir**
## Summary

This PR enhances the `list_network_requests` tool with pagination support to handle large numbers of network requests efficiently. See [this issue](https://github.com/ChromeDevTools/chrome-devtools-mcp/issues/136).

## Motivation

In my experience, the `list_network_requests` tool frequently hits LLM token limits on pages with many requests, making it unusable for modern web applications. I wanted to add pagination to allow agents to be more flexible and manage token limits on their own.

### Pagination Support
- Added `pageSize` parameter to limit requests per call
- Added `pageToken` parameter for navigation between pages
- Added pagination metadata in responses (nextPageToken, previousPageToken, startIndex, endIndex, total)

### Implementation Details
- **New utility**: `src/utils/pagination.ts` - generic pagination function
- **Enhanced McpResponse**: Added pagination options to `setIncludeNetworkRequests()`
- **Updated network tool**: Added pagination parameters to schema
- **Offset-based pagination**: Uses numeric tokens, handles invalid tokens gracefully

## Testing
- Comprehensive test coverage for pagination scenarios
- Tests for first page, subsequent pages, invalid tokens, and edge cases
- All existing tests continue to pass

## Backward Compatibility
- If no pagination parameters are provided, the MCP will return all requests (same as before)

### CODE REVIEW

**Comment by OzTamir** [comment]
@OrKoN this is the first PR out of the two in #107 (with your comments fixed 🙂) - let's get this through and I will build the request filtering on top of this and create an additional PR later.

File: src/tools/network.ts [hunk 1]
```diff
@@ -15,9 +15,27 @@ export const listNetworkRequests = defineTool({
...
+  schema: {
+    pageSize: z
+      .number()
+      .int()
+      .positive()
+      .optional()
+      .describe(
+        'Maximum number of requests to return. When omitted, returns all requests.',
+      ),
+    pageToken: z
```
Comment (OrKoN) [inline src/tools/network.ts]: is there a reason to use a pageToken instead of a pageIdx? and make it non opaque so that client can jump to a particular page? I have done some testing and it seems the client gets often confused if you are asking to fetch a specific page with this API.

Comment (OrKoN) [reply src/tools/network.ts]: alternatively, could we change pageToken to be a number and indicate that it is not opaque and it means the index of the first item of the page?

Comment (OzTamir) [reply src/tools/network.ts]: Fair point - updated to use `pageIdx`.

**Comment by OrKoN** [comment]
Thanks!

**Review by OrKoN — APPROVED** [review]
